test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py

```diff
@@ -273,79 +356,101 @@ class TestSparsePauliOpMethods(QiskitTestCase):
 
     RNG = np.random.default_rng(1994)
 
-    def random_spp_op(self, num_qubits, num_terms):
+    def setUp(self):
+        super().setUp()
+
+        def bind_one(a):
+            parameters = a.parameters
+            return complex(a.bind(dict(zip(parameters, [1] * len(parameters)))))
+
+        self.vbind_one = np.vectorize(bind_one, otypes=[complex])
+
+    def random_spp_op(self, num_qubits, num_terms, params=None):
         """Generate a pseudo-random SparsePauliOp"""
-        coeffs = self.RNG.uniform(-1, 1, size=num_terms) + 1j * self.RNG.uniform(
-            -1, 1, size=num_terms
-        )
+        if params is None:
+            coeffs = self.RNG.uniform(-1, 1, size=num_terms) + 1j * self.RNG.uniform(
+                -1, 1, size=num_terms
+            )
+        else:
+            coeffs = np.array(ParameterVector(params, num_terms))
```

Comment:
It's not obvious to me from the call sites what the `params` argument does and how it affects the output of this function.  It might be clearer to call it `use_parameters=False`, and if `True`, generate a random/unique parameter name to build the `ParameterVector`.  You could use `setUp` to create an infinite iterator like
```python
self.parameter_names = (f"param_{x}" for x in itertools.count())
```
and then have this function do `np.array(ParameterVector(next(self.parameter_names), num_terms))`.  That'll guarantee that no two calls within the same test function get the same named parameters, and will avoid the somewhat mysterious `params=[(None, None, None), "abc"]` stuff in the ddt parameterisation in favour of `use_parameters=[False, True]`.

Reply:
Well, this is excellent. I certainly hadn't thought of such a usage of generator.